src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +159,175 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
     }
 }
 
+void IInferRequestInternal::SetBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    bool all_memory = std::all_of(blobs.begin(), blobs.end(), [](const Blob::Ptr& item) {
+        return item && item->is<MemoryBlob>() && !item->is<RemoteBlob>();
+    });
+    OPENVINO_ASSERT(all_memory,
+                    "set_input_tensors/set_tensors error. Default implementation support only local memory tensors");
+
+    if (blobs.size() == 1) {
+        SetBlob(name, blobs[0]);
+        return;
+    }
+
+    checkBatchedBlobs(name, blobs);
+
+    SetBlobsImpl(name, std::make_shared<BatchedBlob>(blobs));
+}
+
+void IInferRequestInternal::SetBlobsImpl(const std::string& name, const BatchedBlob::Ptr& batched_blob) {
+    IE_THROW(NotImplemented) << "set_input_tensors/set_tensors are not supported by this plugin";
+}
+
+void IInferRequestInternal::checkBatchedBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs) {
+    OPENVINO_ASSERT(!blobs.empty(),
+                    "set_input_tensors/set_tensors can't be called with empty blobs for input '",
+                    name,
+                    "'");
+
+    std::shared_ptr<const ov::op::v0::Parameter> param;
+    const auto& inputs = GetInputs();
+    for (const auto& input : inputs) {
+        if (auto p = std::dynamic_pointer_cast<const ov::op::v0::Parameter>(input)) {
+            if (name == p->get_friendly_name()) {
+                param = p;
+                break;
+            }
+        }
+    }
+    OPENVINO_ASSERT(param, "set_input_tensors/set_tensors error. Parameter '", name, "' is not found");
+    OPENVINO_ASSERT(ov::layout::has_batch(param->get_layout()),
+                    "set_input_tensors/set_tensors can be used only for inputs with N(batch) dimension"
+                    " 'layout' defined. Current layout for '",
+                    name,
+                    "' is ",
+                    param->get_layout().to_string());
+    auto batch_idx = ov::layout::batch_idx(param->get_layout());
+    if (batch_idx < 0) {
+        batch_idx += static_cast<int64_t>(blobs[0]->getTensorDesc().getDims().size());
+    }
+    auto input_size = std::accumulate(blobs.begin(), blobs.end(), 0, [&batch_idx](int res, const Blob::Ptr& item) {
+        OPENVINO_ASSERT(item->getTensorDesc().getDims()[batch_idx] == 1,
```

Comment:
if the dims[batch_idx] should be =1, why we need the std::accumulate the and the inputs_zie is exactly the batch_size (== _inputs[name][batch_idx])

Reply:
In previous versions I accepted base when user passes blobs {1,2,3} for batch=6, that's why I used accumulate. Now updated to 'std::for_each'